### ISSUE TITLE
Add autoplay settings

### ISSUE THREAD

**Issue by mcarare** [mozilla-mobile/focus-android#5743]
Some reviews on the play store are mentioning the lack of setting to disable video/ audio autoplay.

Did we use to have this on the previous versions?

We should consider adding a setting to allow/block autoplay. Maybe not as complex as the one in Fenix, or maybe just disable autoplay by default?

**Comment by brampitoyo** [comment]
@mcarare Focus is a browser where it makes sense to disable autoplay by default. In other words, our default policy should be “Block audio and video”.

**Comment by ionutbedregeanu** [comment]
I took a look in Fenix to see how it was implemented the Autoplay feature and I noticed that we are using SitePermissionsFeature component.
In order to use SitePermissionsFeature in Focus we need to add some additional functionality which is not necessarily helpful because we just want to block everything related to autoplay, without user input

Also from this discussion https://bugzilla.mozilla.org/show_bug.cgi?id=1577596#c18 seems that it doesn't exists a shortcut to set directly in GV the state of Autoplay

@Amejia481 Maybe you have more context about this. 
Should I add a new component in AC in order to be able to block Autoplay directly or should I use SitePermissionsFeature as in Fenix?

**Comment by Amejia481** [comment]
👋🏽 @ionutbedregeanu 

In case, you want to give a global value for a permission you could use `SitePermissionsFeature.sitePermissionsRules`,
with it you can control how a permission should behave globally by default behavior on each page:

For example:
 
If you want to **block** `autoplayAudible` and **allow** `autoplayInaudible` (This is the recommended default value as blocking both could cause some pages to not work properly for more details see https://github.com/mozilla-mobile/fenix/issues/16557#issuecomment-729820241).
 
```kotlin
val sitePermissionsRules = SitePermissionsRules(
            notification = Action.ASK_TO_ALLOW,
            microphone = Action.ASK_TO_ALLOW,
            location = Action.ASK_TO_ALLOW,
            camera = Action.ASK_TO_ALLOW,
            autoplayAudible = AutoplayAction.BLOCKED,
            autoplayInaudible = AutoplayAction.ALLOWED,
            persistentStorage = Action.ASK_TO_ALLOW,
            mediaKeySystemAccess = Action.ASK_TO_ALLOW
)

SitePermissionsFeature(
  ..., // Other properties
  sitePermissionsRules = sitePermissionsRules
  , ... // Other properties
)
```
You can find an example on Fenix [here](https://github.com/mozilla-mobile/fenix/blob/f8adaae5cae6c88b4513d4823cb5aed9017087a1/app/src/main/java/org/mozilla/fenix/utils/Settings.kt#L923) :) 

**Be ware** that the values that you are indicating value for other permissions will affect how these permissions work, if you want that they continue working as they are requesting user confirmation just use `ASK_TO_ALLOW`.

If you any additional help just let me know happy to do so :)

**Comment by ionutbedregeanu** [comment]
Note for QA: As Arturo said we can't set as blocked at this moment both permissions, autoplayAudible and autoplayInaudible, so I set only the autoplayAudible

**Comment by LaurentiuApahideanSV** [comment]
I tested the issue on the latest debug build from main 2021-11-11.
I tested the following websites:
https://googlechrome.github.io/samples/muted-autoplay/
https://pauljadam.com/demos/autoplay-loop-muted-controls.html
https://video-dev.github.io/can-autoplay/
https://www.wired.com/story/watch-a-robot-dog-learn-how-to-deftly-fend-off-a-human/amp
Video with audio will not start auto playing only without audio will start auto playing.

### Devices used:
- Oppo Find X3 (Android 11)
- OnePlus 6T (Android 9)

**Comment by delia-pop** [comment]
This issue is occurring on the 96.0.0-beta.1 (Build 353412232 GV 96.0-20211206194555) build, RC 95.1.0 and latest debug build from main 12/08. The autoplay starts even when the video has audio enabled. 

Tested with the above websites:
https://googlechrome.github.io/samples/muted-autoplay/
https://pauljadam.com/demos/autoplay-loop-muted-controls.html
https://video-dev.github.io/can-autoplay/
https://www.wired.com/story/watch-a-robot-dog-learn-how-to-deftly-fend-off-a-human/amp

Devices used:
* Samsung Galaxy Note10 (Android 11)
* LG G7 fit (Android 8.1.0)
* Xiaomi Redmi Note 8T (Android 9)

**Comment by ionutbedregeanu** [comment]
> 

I tested with Pixel 3 Api 30 on all the metioned websites and everything seems ok on my side except the fact that I had to refresh the page manually to apply the new autoplay option. I will make a pr for that

**Comment by lobontiumira** [comment]
Verified as fixed on the Focus Nightly build 98.0 (GV 98.0a1-20220111093827) on the following pages:
* https://googlechrome.github.io/samples/muted-autoplay/
* https://pauljadam.com/demos/autoplay-loop-muted-controls.html

with Lenovo Tab M10 (Android10), Oppo Find X3 Lie (Android 11), HTC 10 (Android 8), and Sony Xperia Z5 Premium (Android 7.1.1).
Changing the autoplay settings take effect without refreshing the pages.
